t/04-commands-implemented.t

```diff
@@ -1,36 +1,56 @@
 use strict;
 use warnings;
 
-# TODO: find another way to do this checking, this is so fragile
-use Selenium::Remote::Commands;
-use Test::More;
+use Test::More tests => 3 + 1;
+use Test::NoWarnings;
 
-unless($ENV{RELEASE_TESTING}) {
-  plan(skip_all=>"Author tests not required for installation.");
+# Assume we're not using dzil test unless we're doing that
+if( !grep { index( $_, ".build" ) != -1 } @INC ) {
+  require Cwd;
+  require File::Basename;
+  push( @INC, File::Basename::dirname(Cwd::abs_path(__FILE__)) . "/../lib" );
 }
 
-my $comm = Selenium::Remote::Commands->new->get_cmds;
-for my $command (keys %{$comm}) {
-  my $found_command = 0;
-  for my $file (
-    qw{lib/Selenium/Remote/Driver.pm
-    lib/Selenium/Remote/WebElement.pm
-    lib/Selenium/Firefox.pm}
-    ) {
-    open(my $fh, '<', $file) or die "Couldn't open file $file";
-    for (<$fh>) {
-      if (/'?command'?\s*=>\s*'$command'/
-       or /{'?commands'?}->\{'?$command'?}/) {
-        pass("find $command");
-        $found_command = 1;
+require_ok( "Selenium::Remote::Commands" ) || die;
+
+subtest "All implemented commands are found" => sub {
+  plan skip_all => "Author tests not required for installation." unless $ENV{'RELEASE_TESTING'};
+  my $comm = Selenium::Remote::Commands->new->get_cmds;
+  for my $command (keys %{$comm}) {
+    my $found_command = 0;
+    for my $file (
+      qw{lib/Selenium/Remote/Driver.pm
+      lib/Selenium/Remote/WebElement.pm
+      lib/Selenium/Firefox.pm}
+      ) {
+      open(my $fh, '<', $file) or die "Couldn't open file $file";
+      for (<$fh>) {
+        if ( /'?command'?\s*=>\s*'$command'/ or /{'?commands'?}->\{'?$command'?}/) {
+          pass("find $command");
+          $found_command = 1;
+        }
       }
     }
+    if (!$found_command && $command !~ /Gecko/) {
```

Comment:
This seems profoundly silly to use pass/fail when one could simply fall through and do ok($found_command,"found $command") unless $command =~ /Gecko/

Reply:
I don't recall writing that bit... maybe I moved it?